components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/OperationPolicyProvider.java

```diff
@@ -0,0 +1,208 @@
+     * @throws APIManagementException failed to get API-Specific operation policy
+     */
+    OperationPolicyData getAPISpecificOperationPolicyByPolicyID
+    (String apiUUID, String policyId, String organization, boolean isWithPolicyDefinition)
+            throws APIManagementException;
+
+    /**
+     * Get all API-Specific operation policies of an API. (these are not attached policies to the API)
+     *
+     * @param apiUUID      API UUID
+     * @param organization organization
+     * @return List of Operation policy data objects
+     * @throws APIManagementException failed to get API-Specific operation policies
+     */
+    List<OperationPolicyData> getAllAPiSpecificOperationPolicies(String apiUUID, String organization)
+            throws APIManagementException;
+
+    /**
+     * Clone an API-Specific policy from one API to another API as API-Specific policy
+     *
+     * @param policyId     policy ID
+     * @param OldApiUUID   API UUID (from)
+     * @param NewApiUUID   API UUID (to)
+     * @param organization organization
+     * @return cloned policy id
+     * @throws APIManagementException failed to clone operation policy
+     */
+    String cloneAPISpecificOperationPolicy(String policyId, String OldApiUUID, String NewApiUUID, String organization)
+            throws APIManagementException;
+
+    /**
+     * Clone all operation policies from one API to another API as API-Specific policies
+     *
+     * @param oldApiUUID   API UUID (from)
+     * @param newApiUUID   API UUID (to)
+     * @param organization organization
+     * @throws APIManagementException failed to clone operation policies
+     */
+    void cloneAllApiSpecificOperationPolicies(String oldApiUUID, String newApiUUID, String organization)
+            throws APIManagementException;
+
+    /**
+     * Revision API-Specific operation policy.
+     *
+     * @param policyId     policy ID
+     * @param apiUUID      API UUID
+     * @param revisionId   revision ID of policy
+     * @param organization organization
+     * @return revisioned policy ID
+     * @throws APIManagementException failed to revision operation policy
+     */
+    String revisionOperationPolicy(String policyId, String apiUUID, String revisionId, String organization)
+            throws APIManagementException;
+
+    /**
+     * Add operation policy to an API
+     *
+     * @param apiID               API UUID
+     * @param operationPolicyData policy data
+     * @param organization        organization
+     * @return newly added policy ID
+     * @throws APIManagementException failed to add API-Specific policy
+     */
+    String addAPISpecificOperationPolicy(String apiID, OperationPolicyData operationPolicyData, String organization)
+            throws APIManagementException;
+
+    //Other
+
+    /**
+     * This is Choreo specific function.
+     * Returns true in Choreo context, false otherwise.
+     *
+     * @return bool
+     */
+    Boolean isReadOnly();
```

Comment:
I think it's for future use. to indicate BCentral Policies in the Choreo context are read-only.

Reply:
In that case, can we be a bit more explicit about what we are checking here? Because when reading this method name, I don't get the impression that it's actually referring to the access mode of the policies themselves.

Reply:
If we are not using this, lets remove this function from the interface. @ThusharaSampath